src/app/core/tools/handlers/CopyHandler.ts

```diff
@@ -27,7 +27,12 @@ export const CopyHandler: EventHandler = ({
         // We don't copy the data from the json since it will cause 
         // some weird error, which will cause the issue #746
         ev.clipboardData.setData("text/plain", str);
-        ev.preventDefault(); // Necessary to copy correctly
+
+        // If no text is selected, copy the component
+        // Otherwise, do default copying
```

Comment:
Can you expand this comment to explain what exactly this if-statement is doing?
I'm not sure what the `'LABEL'` thing represents, or why the `"CARET"` matters?

Also, you should say (in the comment) that this statement is necessary because it "fixes #874"

Reply:
Yeah that definitely wasn't clear, hopefully the comments I added make it more understandable.